### ISSUE TITLE
WireMock - invocations verification in consecutive tests fails

### ISSUE THREAD

**Issue by szpak** [4finance/micro-infra-spring#186]
How to reproduce:
1. Duplicate `should find a place by verifying tweet's geolocation` test in boot-microservice.
2. Second test fails with:

```
Requests received: [ ]
	at com.github.tomakehurst.wiremock.client.WireMock.verifyThat(WireMock.java:248)
	at com.ofg.twitter.places.AcceptanceSpec.should find a place by verifying tweet's coordinates2(AcceptanceSpec.groovy:150)
```

WireMock is prepared in MvcWiremockIntegration:
```
    void setup() {
        wireMock = new WireMock(wiremockUrl, httpMockServer.port())
        wireMock.resetToDefaultMappings()
    }
```

**Comment by mchmielarz** [comment]
This seems to be a valid behaviour of `WireMock`.

The test extends `MvcWiremockIntegrationSpec` class where in `setup()` method all mappings are reset to default ones. This means sending such request to `HttpMockServer` where all mappings are stored. Because of that in the second run there are no defined mappings for the test case.

This can be avoided by calling static `WireMock.saveAllMappings()` method which stores defined mappings to a file. After this when mock is reset to default settings previously saved mappings are loaded from this file.

So in summary, `WireMock` clears out all mappings because there are defined no default mappings.

[Here](https://github.com/tomakehurst/wiremock/blob/master/src/test/java/com/github/tomakehurst/wiremock/SavingMappingsAcceptanceTest.java) are tests of this feature.

**Comment by mchmielarz** [comment]
@szpak I've checked mappings registration in `WireMock` in the troubling test. So reset has nothing to do with the behaviour you've described.

Going deeper in the code of the `boot-microservice` project I've found the [CityFinder](https://github.com/4finance/boot-microservice/blob/master/src/main/groovy/com/ofg/twitter/place/extractor/CityFinder.groovy) class has `findCityFromCoordinates` method annotated with `@Cacheable`.

Since the method was called in previous test with exactly the same arguments it won't be executed in the second test. The result returned back from it will come from Spring's internal cache and `WeatherClient` will not be called to connect to remote service. This is why `WireMock` complains at the end of the copied test. :)

I've checked tests with caching disabled and both work as expected.

**Comment by szpak** [comment]
The simplest solutions are the best... I'm glad that you had found it.

**Comment by marcingrzejszczak** [comment]
I'm sorry that I've made your life miserable with caching that method :P I've totally forgotten about it... :/
